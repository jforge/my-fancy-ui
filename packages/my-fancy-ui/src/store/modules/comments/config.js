// @flow

export const namepsace = '@immowelt/my-fancy-ui';
export const moduleId = 'comments';
