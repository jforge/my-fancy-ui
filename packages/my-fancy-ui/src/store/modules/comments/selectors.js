// @flow

import type {StateType} from './../../types.js';
import type {CommentType, CommentIdType, CommentsByIdType} from './types.js';

import {$get} from 'plow-js';
import {moduleId} from './config.js';

export const getCommentsById = (state: StateType): CommentsByIdType =>
  $get([moduleId, 'commentsById'], state);
export const getCommentForId = (
  state: StateType,
  props: {id: CommentIdType}
): CommentType => $get([moduleId, 'commentsById', props.id], state);
