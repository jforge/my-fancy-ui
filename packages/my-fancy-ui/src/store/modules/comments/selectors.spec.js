import * as selectors from './selectors.js';
import {moduleId} from './config.js';

describe('getCommentsById()', () => {
  it('should be a function.', () => {
    expect(typeof selectors.getCommentsById).toBe('function');
  });

  it(`should return the "${moduleId}.commentsById" value.`, () => {
    const state = {
      [moduleId]: {
        commentsById: 'foo'
      }
    };
    const result = selectors.getCommentsById(state);

    expect(result).toMatchSnapshot();
  });
});

describe('getCommentForId()', () => {
  it('should be a function.', () => {
    expect(typeof selectors.getCommentForId).toBe('function');
  });

  it(`should return the value of the "${moduleId}.commentsById" key which matches the propagated prop id.`, () => {
    const state = {
      [moduleId]: {
        commentsById: {
          foo: 'bar'
        }
      }
    };
    const props = {
      id: 'foo'
    };
    const result = selectors.getCommentForId(state, props);

    expect(result).toMatchSnapshot();
  });
});
