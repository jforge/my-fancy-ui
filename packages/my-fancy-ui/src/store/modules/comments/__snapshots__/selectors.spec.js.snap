// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCommentForId() should return the value of the "comments.commentsById" key which matches the propagated prop id. 1`] = `"bar"`;

exports[`getCommentsById() should return the "comments.commentsById" value. 1`] = `"foo"`;
